### DISCUSSION TITLE
同时在多个PagView里播放动效，此场景内存使用会明显增长，请问替换为PAGImagView可以改善内存消耗吗

### DISCUSSION THREAD

**Discussion by camp1990** [Tencent/libpag#1625] [Q&A] [answered]
目前的业务场景是：用户点击按钮的时候会播放一个动效，如果点击速度足够快，那么此时会创建多个pagView来展示动效，内存消耗很大，我用了缓存pagView的方式减少创建数量还是会占用很大内存。
请问：
1.此种场景使用PAGImagView替换PagView会降低内存吗
2.创建PAGImagView会消耗内存来创建GPU缓冲区吗

**Comment by kevingpqi123** [comment, marked as answer]
可以阅读下这篇文章:https://pag.art/docs/use-pagimageview.html

**Reply by camp1990** [reply]
文中提到“单个动效文件在整个生命周期中用户只会看到一次短暂基于 GPU 实时渲染的画面，后续都是直接读取高速的磁盘缓存来呈现"，只会看到一次短暂基于 GPU 实时渲染,也就是说首次播放某个pag文件，创建载体PAGImageView还是会有一定内存消耗是吗，请问有具体数据吗

**Reply by kevingpqi123** [reply]
你可能理解有些偏差，对于 pag 而言，本身就是 GPU 渲染的方案，无论那种方案 PAGView 还是 PAGImageView， 都需要先将动效渲染出内容，渲染的过程中就是需要使用到 GPU 渲染环境，差异的是 PAGImageView 的方案渲染的同时会将渲染数据缓存到本地，一旦缓存完成，就会将 GPU 环境销毁掉，后续就是本地缓存数据的读取显示，不再需要 GPU 环境
关于性能数据，你直接到 pag 的 demo 工程中测试就可以获取到

**Reply by camp1990** [reply]
好的，感谢解答疑问
